lib/demos/reference/motion_demo_fade_scale_transition.dart

```diff
@@ -0,0 +1,158 @@
+
+  bool get _isAnimationRunningForwardsOrComplete {
+    switch (_controller.status) {
+      case AnimationStatus.forward:
+      case AnimationStatus.completed:
+        return true;
+      case AnimationStatus.reverse:
+      case AnimationStatus.dismissed:
+        return false;
+    }
+    assert(false);
+    return null;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final localizations = GalleryLocalizations.of(context);
+    final colorScheme = Theme.of(context).colorScheme;
+
+    return Scaffold(
+      appBar: AppBar(
+        automaticallyImplyLeading: false,
+        title: Text(localizations.demoFadeScaleTitle),
+      ),
+      floatingActionButton: AnimatedBuilder(
+        animation: _controller,
+        builder: (context, child) {
+          return FadeScaleTransition(
+            animation: _controller,
+            child: child,
+          );
+        },
+        child: Visibility(
+          visible: _controller.status != AnimationStatus.dismissed,
+          child: FloatingActionButton(
+            child: const Icon(Icons.add),
+            onPressed: () {},
+          ),
+        ),
+      ),
+      bottomNavigationBar: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          const Divider(height: 0),
+          Padding(
+            padding: const EdgeInsets.symmetric(vertical: 8),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                RaisedButton(
+                  onPressed: () {
+                    showModal<void>(
+                      context: context,
+                      builder: (context) {
+                        return _ExampleAlertDialog();
+                      },
+                    );
+                  },
+                  color: colorScheme.primary,
+                  textColor: colorScheme.onPrimary,
+                  child: Text(localizations.demoFadeScaleShowAlertDialogButton),
+                ),
+                const SizedBox(width: 10),
+                RaisedButton(
+                  onPressed: () {
+                    if (_isAnimationRunningForwardsOrComplete) {
+                      _controller.reverse();
+                    } else {
+                      _controller.forward();
+                    }
+                  },
+                  color: colorScheme.primary,
+                  textColor: colorScheme.onPrimary,
+                  child: Text(
+                    _isAnimationRunningForwardsOrComplete
+                        ? localizations.demoFadeScaleHideFabButton
+                        : localizations.demoFadeScaleShowFabButton,
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _ExampleAlertDialog extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final localizations = GalleryLocalizations.of(context);
+
+    return AlertDialog(
```

Comment:
In the gif it looks like the dialog doesn't follow the demo theme, I remember running into this when I did the dialog demo and resolved it by wrapping the widget with the theme, like here: https://github.com/flutter/gallery/blob/master/lib/demos/material/dialog_demo.dart#L41

Reply:
Thanks I'll check this out.